### PR TITLE
Support Fixed key

### PR DESCRIPTION
**Pull request by solsson**
I use kafkacat in replicated containers, where a useful key is for example the machine that the particular container runs on, which I can get from environment variables. It makes more sense to set this key value at command invocation, than `sed` ing the piped data.

I remember nothing of the C we wrote in school, but I gave this a shot. If you like the idea but - as is likely - not the implementaton or the choice of flag, I'm happy to see a rewrite.

I've built this as [solsson/kafkacat-fixedkey](https://hub.docker.com/r/solsson/kafkacat-fixedkey/) and tested it briefly for my usecases, including those with per-message keys from stdin.

### CODE REVIEW

File: kafkacat.c [hunk 1]
```diff
@@ -350,6 +351,11 @@ static void producer_run (FILE *fp, char **paths, int pathcnt) {
                                 }
                         }
 
+                        if (!key && conf.fixed_key) {
+                                key = conf.fixed_key;
+                                key_len = (size_t)(strlen(conf.fixed_key));
```
Comment (edenhill) [inline kafkacat.c]: Cache key_len on conf

File: kafkacat.c [hunk 2]
```diff
@@ -936,6 +942,7 @@ static void RD_NORETURN usage (const char *argv0, int exitcode,
                 "  -p -1              Use random partitioner\n"
                 "  -D <delim>         Delimiter to split input into messages\n"
                 "  -K <delim>         Delimiter to split input key and message\n"
+                "  -F <str>           Use a fixed key for all messages\n"
```
Comment (edenhill) [inline kafkacat.c]: I think `-k` would be more intuitive

Comment (solsson) [reply kafkacat.c]: Agreed. Funny I didn't think of that alternative.

File: kafkacat.c (hunk 1 above)
Comment (solsson) [reply kafkacat.c]: Good idea. Will do.

**Comment by solsson** [comment]
I've been running the build from after my review fixes, and see no regressions.

**Comment by edenhill** [comment]
Thank you for this!

**Pull request merged**